zerver/models/messages.py

```diff
@@ -670,6 +670,9 @@ class ImageAttachment(models.Model):
     original_height_px = models.IntegerField()
     frames = models.IntegerField()
 
+    # indicates whether we attempted to delete this image
+    deleted = models.BooleanField(default=False)
```

Comment:
These should probably have `db_index=True`, so it's easy to query them without a table scan.